src/mcts/search.cc

```diff
@@ -1444,16 +1449,14 @@ void SearchWorker::FetchSingleNodeResult(NodeToProcess* node_to_process,
     // Note that we want to calculate (exp(p-max_p))^(1/T) = exp((p-max_p)/T).
     p = FastExp((p - max_p) / params_.GetPolicySoftmaxTemp());
 
-    // Note that p now lies in [0, 1], so it is safe to store it in compressed
-    // format. Normalization happens later.
-    edge.edge()->SetP(p);
-    // Edge::SetP does some rounding, so only add to the total after rounding.
-    total += edge.edge()->GetP();
+    intermediate[counter++] = p;
+    total += p;
```

Comment:
Looks like this resulted in some accuracy improvements for fp16 in addition to the speed optimization. @fohristiwhirl noticed:
```
9.02, 9.01, 7.62, 6.33, 5.82 fp16 0.25.1
9.01, 9.00, 7.62, 6.32, 5.85 fp16 master
9.00, 8.97, 7.59, 6.32, 5.86 my opencl/blas
```
These are the top 5 policies using sv384x30-swa-3010000 from
```
position startpos moves d2d4 g8f6 g1f3 e7e6 c1f4 d7d5 e2e3 c7c5 b1d2 d8b6 a1b1 c5d4 e3d4 c8d7 c2c3 d7b5 d1b3 b5f1 b3b6 a7b6 h1f1 b8c6 a2a3 f6h5 f4g3 f7f6 e1d1 g7g5 d1c2 a8c8 d2b3 h5g7 h2h3 g7f5 g3h2 h7h5 b1e1 e8f7 c2b1 g5g4 f3d2 f8h6 e1e2 h8g8 h3g4 g8g4 g2g3 c6a5 f1e1 c8e8 b1c2 a5c6 d2f3 h6f8 b3c1 f8d6 e1d1 g4e4 d1e1 e4e2 e1e2 e6e5 d4e5 f6e5 c2d1 e5e4 f3d2 c6e5 d1c2 d6c5 a3a4
```